### DISCUSSION TITLE
Set up CSL (aplic WDP 303075) or find a template

### DISCUSSION THREAD

**Discussion by pctelco** [arendst/Tasmota#10403] [Q&A] [answered]
Hello everybody,
Unfortunately, my English is not good enough to have a conversation here in English (mother tongue is German), so I use google-translate to write and apologize for this in advance.
Among other things, I read here in the thread ("adding CSL (aplic WDP 303075) Socket # 3991", https://github.com/arendst/Tasmota/issues/3991), which is unfortunately closed, that people here have managed to get the socket running, including the display of the values, however Unfortunately, reading there does not reveal to me how this succeeded.
I own two of these switchable sockets, along with other other types with a matching template, and have flashed the first of the two mentioned here with Tasmota 9.2.0 OTA and running as "generic (18)" as described there as a pure switch but I would be happy if someone could help me to activate the sensors. Unfortunately, I couldn't find a suitable template anywhere.

**Comment by barbudor** [comment, marked as answer]
Hello
Did you read this page from the docs : https://tasmota.github.io/docs/devices/CSL-aplic-WDP-303075/

**Reply by barbudor** [reply]
OK, there is nothing new here
Let me search more

**Reply by barbudor** [reply]
You need to select the Aplic module

Go to Configuration => Configure Template
Configure as follow :
![image](https://user-images.githubusercontent.com/2996042/103560394-fd43e900-4eb7-11eb-87d9-997b0a83bfe0.png)
![image](https://user-images.githubusercontent.com/2996042/103560447-1482d680-4eb8-11eb-803b-eadfbb338abd.png)

**Reply by barbudor** [reply]
If the above doesn't work as expect, can you please add the following:
GPIO04 : HLW8012 CF
GPIO05 : HLW8012 CF1
GPIO12 : HLW8012 CF Sel

Thanks for letting us know when it works so we can properly update the documentation

**Reply by pctelco** [reply]
Thank you very much! This is how it worked now, without having to adjust the entries for GPIO04, GPIO05 and GPIO12.
After calibrating several times, I now even have the power factor stable at 1.0 under load. Before calibration, all values were very far from reality. This showed me that I should recalibrate my other 5 switchable sockets with power measurement in order to get more precise values, because there a power factor of 0.85 to 0.95 is displayed.

**Reply by barbudor** [reply]
Cool 👍 
Can you please type the command `Template` in the web console and copy back the result here ?
Also please provide a link where you purchased the unit so we can update the Template database
Thanks

**Reply by pctelco** [reply]
The result of the "Template" command is {"NAME": "CSL Aplic", "GPIO": [0,0,0,32,2688,2656,0,0,2624,320,224,0,0,0], "FLAG": 0, "BASE": 53}.
I bought the adapter (it was a two-pack) over 2 years ago on Amazon (Link: https://www.amazon.de/gp/product/B07J5MXCJK/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1), but you can I haven't bought it there for a long time and the offer page looks very strange today.
